File: use_cases/common_cases/include/c002_cpp_structs.cpp
#include "c002_cpp_structs.h"

const vec2_t vec2_t::sZero = { 0, 0 };

vec2_t vec2_t::add(vec2_t other) const {
    return vec2_t { this->x + other.x, this->y + other.y };
}

File: use_cases/common_cases/c002_cpp_structs.zig
pub const vec2_t = extern struct {
    extern const _ZN6vec2_t5sZeroE: vec2_t;
    pub inline fn sZero() vec2_t {
        return _ZN6vec2_t5sZeroE;
    }

    x: f32,
    y: f32,

    extern fn _ZNK6vec2_t3addES_(self: *const vec2_t, other: vec2_t) vec2_t;
    pub const add = _ZNK6vec2_t3addES_;
};

pub const Vec2 = vec2_t;
